airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/ResourceRequirementsUtils.java

```diff
@@ -0,0 +1,103 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+public class ResourceRequirementsUtils {
+
+  /**
+   * Given connection-level resource requirements, actor-definition-level resource requirements,
+   * worker-default resource requirements, and a job type, returns the final resource requirements
+   * generated by merging the provided requirements in hierarchy order.
+   *
+   * Connection-level resource requirements take precendence over actor-definition level resource
+   * requirements. Within the actor-definition level requirements, job-type-specific requirements take
+   * precedence over default definition requirements. Actor-definition level resource requirements
+   * take precedence over worker default resource requirements.
+   *
+   * @param connectionResourceReqs - the resource requirements set on the connection
+   * @param actorDefinitionResourceReqs - the resource requirements set on the actor definition
+   * @param workerDefaultResourceReqs - the default worker resource requirements set in the env
+   *        variables
+   * @param jobType - type of job to extract resource requirements for from the actor definition reqs
+   * @return resource requirements, if present, otherwise an empty ResourceRequirements object.
+   */
+  public static ResourceRequirements getResourceRequirements(@Nullable final ResourceRequirements connectionResourceReqs,
+                                                             @Nullable final ActorDefinitionResourceRequirements actorDefinitionResourceReqs,
+                                                             @Nullable final ResourceRequirements workerDefaultResourceReqs,
+                                                             final JobType jobType) {
+    final ResourceRequirements jobSpecificDefinitionResourceReqs = getResourceRequirementsForJobType(actorDefinitionResourceReqs, jobType)
+        .orElse(null);
+    final ResourceRequirements defaultDefinitionResourceReqs = Optional.ofNullable(actorDefinitionResourceReqs)
+        .map(ActorDefinitionResourceRequirements::getDefault).orElse(null);
+    return mergeResourceRequirements(
+        connectionResourceReqs,
+        jobSpecificDefinitionResourceReqs,
+        defaultDefinitionResourceReqs,
+        workerDefaultResourceReqs);
+  }
+
+  /**
+   * Given a list of resource requirements, merges them together. Earlier reqs override later ones.
+   *
+   * @param resourceReqs - list of resource request to merge
+   * @return merged resource req
+   */
+  public static ResourceRequirements mergeResourceRequirements(final ResourceRequirements... resourceReqs) {
+    final ResourceRequirements outputReqs = new ResourceRequirements();
+    final List<ResourceRequirements> reversed = new ArrayList<>(Arrays.asList(resourceReqs));
+    Collections.reverse(reversed);
+
+    for (final ResourceRequirements resourceReq : reversed) {
```

Comment:
might be worth a quick comment explaining the strategy, ie,
```
// start from the lowest priority requirements so that we can repeatedly override the output requirements to guarantee that we end with the highest priority setting for each
```